conan/tools/build/__init__.py

```diff
@@ -19,11 +19,36 @@ def use_win_mingw(conanfile):
     return False
 
 
-def args_to_string(args):
+def cmd_args_to_string(args):
     if not args:
         return ""
-    # FIXME: This is ugly, hardcoding condition how to parse args
     if sys.platform == 'win32':
-        return subprocess.list2cmdline(args)
+        return _windows_cmd_args_to_string(args)
     else:
-        return " ".join("'" + arg.replace("'", r"'\''") + "'" for arg in args)
+        return _unix_cmd_args_to_string(args)
+
+
+def _unix_cmd_args_to_string(args):
+    """Return a shell-escaped string from *split_command*."""
+    return ' '.join(quote(arg) for arg in args)
+
+
+def _windows_cmd_args_to_string(args):
```

Comment:
NOTE: This is doing more or less the same than `list2cmdline` but the original implementation is a little bit more complex (not too much) and covers more cases with "escapes of escapes of escaped quotes". I would say it would be convenient to just copy-paste the python original code.